### PR TITLE
Update setup instructions in README

### PR DESCRIPTION
**Pull request by r-richardson**
This PR updates the setup documentation and adds the minimal postgres authentication config necessary for startup.

Created as part of [163586](https://progress.opensuse.org/issues/163586), as when asking for help my team member ran into the same issues during setup as i did.

### CODE REVIEW

**Comment by r-richardson** [comment]
> @r-richardson Please mention the related ticket

yes sorry, i created this as part of [163586](https://progress.opensuse.org/issues/163586), as when asking for help my team member ran into the same issues during setup as i did.

**Comment by kraih** [comment]
Please don't put everything in the README. We can move `API.md` and any new documentation into a `docs` folder.

**Comment by kraih** [comment]
I've just started adding more docs for Cavil, maybe it can be organised similarly here. https://github.com/openSUSE/cavil/tree/master/docs

**Comment by perlpunk** [comment]
The ticket link should go into the commit message and the PR description

File: README.md [hunk 1]
```diff
@@ -9,14 +9,53 @@ which incidents. Its deployment happens once a day via
...
+host    postgres        postgres        127.0.0.1/32            trust
+host    all             all             127.0.0.1/32            reject
+# IPv6 local connections:
+host    postgres        postgres        ::1/128                 trust
+host    all             all             ::1/128                 reject
+# Allow replication connections from localhost, by a user with the
+# replication privilege.
+local   replication     all                                     reject
+host    replication     all             127.0.0.1/32            reject
+host    replication     all             ::1/128                 reject
```
Comment (okurz) [inline README.md:55]: Do all of those lines need changes or can you focus on just the parts that needed changes?

File: README.md [hunk 2]
```diff
@@ -61,6 +99,10 @@ For a local PostgreSQL instance with a username and password one could call:
 ```
 TEST_ONLINE=postgresql://postgres:postgres@localhost:5432/postgres prove -l t/*.t t/*.t.js
 ```
+
+### Documentation
+For further documentation, please see **[docs](https://github.com/openSUSE/qem-dashboard/tree/main/docs)** (WIP)
```
Comment (okurz) [inline README.md:104]: What's the "WIP" about? Isn't everything WIP which isn't carved into stone? :)

**Review by okurz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (r-richardson) [reply README.md:55]: Yes and no, i googled and it seems that the default `METHOD` values are not consistent, which is why i just set all defaults to `reject` and added only the specific permissions needed for running the qem-dashboard (line 46 and 49). Its also important that the settings are in the correct order. So in theory adding these two lines would be enough, though less secure.

File: README.md (hunk 2 above)
Comment (r-richardson) [reply README.md:104]: true :+1: I'll remove it

**Review by okurz — DISMISSED** [review]

**Review by kalikiana — APPROVED** [review]
Nice!

**Review by Martchus — CHANGES_REQUESTED** [review]
The commit from @dependabot should not be on this branch. Do an interactive rebase against the main branch and drop this commit completely from the list of commits. This will hopefully also remove the conflict.

File: README.md [hunk 3]
```diff
@@ -9,14 +9,53 @@ which incidents. Its deployment happens once a day via
...
+
+if you are on a zypper-based system, omit the `--with-deps` parameter
+
+    npx playwright install
+
+and ignore the missing dependencies
+
+#### nodejs18
+
+In case `nodejs18` is not available (e.g. Tumbleweed), you can alternatively get it via the [Node Version Manager](https://github.com/nvm-sh/nvm). Simply [install nvm](https://github.com/nvm-sh/nvm?tab=readme-ov-file#install--update-script) then run `nvm install 18` and `nvm use 18`.
```
Comment (kalikiana) [inline README.md:33]: What is this needed for? `npm install` worked fine for me on Tumbelweed.

Comment (Martchus) [reply README.md:33]: Good point. `npm install` works also just fine for me. There is no need to install version 18 specifically.

File: README.md [hunk 4]
```diff
@@ -9,14 +9,53 @@ which incidents. Its deployment happens once a day via
...
-    $ sudo zypper in -C postgresql-server postgresql-contrib
-    $ sudo zypper in -C perl-Mojolicious perl-Mojolicious-Plugin-Webpack \
-      perl-Mojo-Pg perl-Cpanel-JSON-XS perl-JSON-Validator perl-IO-Socket-SSL nodejs16
-    $ npm install
+To install all required dependencies, run
 
-Update the config file `dashboard.yml` to point to your PostgreSQL database (and other services where appropriate):
+    sudo zypper in -C postgresql-server postgresql-contrib
+    sudo zypper in -C perl-Mojolicious perl-Mojolicious-Plugin-Webpack \
+      perl-Mojo-Pg perl-Cpanel-JSON-XS perl-JSON-Validator perl-IO-Socket-SSL nodejs18
```
Comment (Martchus) [inline README.md:18]: We don't need to install an old version of nodejs. The current/default version works fine. I'd explicitly mention npm, though:
```suggestion
            perl-Mojo-Pg perl-Cpanel-JSON-XS perl-JSON-Validator perl-IO-Socket-SSL nodejs npm
```

File: README.md [hunk 5]
```diff
@@ -9,14 +9,53 @@ which incidents. Its deployment happens once a day via
...
+
+    npx playwright install --with-deps
+
+if you are on a zypper-based system, omit the `--with-deps` parameter
+
+    npx playwright install
+
+and ignore the missing dependencies
+
+#### nodejs18
```
Comment (Martchus) [inline README.md:31]: I would drop this whole paragraph. We don't need a specific nodejs version.

File: README.md (hunk 3 above)
Comment (kraih) [reply README.md:33]: Should be 18 or newer.

File: assets/vue/App.vue [hunk 6]
```diff
@@ -31,7 +31,11 @@
           </ul>
           <ul class="navbar-nav flex-row flex-wrap ms-md-auto" id="navbarAPI">
             <li class="nav-item">
-              <a class="nav-link" href="https://github.com/openSUSE/qem-dashboard/blob/main/API.md" target="_blank">
+              <a
+                class="nav-link"
+                href="https://github.com/openSUSE/qem-dashboard/blob/main/docs/API.md"
```
Comment (Martchus) [inline assets/vue/App.vue:36]: This URL returns a 404 page.

File: docs/folders.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+- **[migrations](https://github.com/openSUSE/qem-dashboard/blob/main/migrations)**: Database migration scripts.
+- **[node_modules](https://github.com/openSUSE/qem-dashboard/tree/main/node_modules)**: Dependencies installed by npm (Node Package Manager)
+- **[package.json](https://github.com/openSUSE/qem-dashboard/tree/main/package.json)** and **[package-lock.json](https://github.com/openSUSE/qem-dashboard/tree/main/package-lock.json)**: Configuration files used to manage dependencies.
+- **[public](https://github.com/openSUSE/qem-dashboard/tree/main/public)**: Publicly accessible assets, such as images and static files.
+- **[script](https://github.com/openSUSE/qem-dashboard/tree/main/script)**: Used to automate tasks and workflows.
+- **[t](https://github.com/openSUSE/qem-dashboard/tree/main/t)**: Containing test files.
+- **[templates](https://github.com/openSUSE/qem-dashboard/tree/main/templates)**: Template files, used to generate dynamic content.
+- **[webpack.config.js](https://github.com/openSUSE/qem-dashboard/tree/main/webpack.config.js)**: Configuration file for Webpack, a JavaScript bundler used to prepare code for production.
```
Comment (Martchus) [inline docs/folders.md:13]: I think it would be possible to make those relative paths.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (r-richardson) [reply README.md]: currently the README says one should install nodejs16, which is not available on TW. As production was using nodejs18 when i first edited the docs, i added the section on how to install that version on TW using nvm. I have now removed that paragraph and added nodejs22 (also available on TW), to the zypper install line in the setup instructions.

File: README.md (hunk 4 above)
Comment (r-richardson) [reply README.md]: I've added `npm`, however as there is no `nodejs` package, i added `nodejs22`, alternatively we could use`nodejs-default`.

File: assets/vue/App.vue (hunk 6 above)
Comment (r-richardson) [reply assets/vue/App.vue:36]: The linked .md file is currently still located in the root directory and is moved to the docs dir, fixing this link, as part of the PR

File: docs/folders.md (hunk 7 above)
Comment (r-richardson) [reply docs/folders.md]: Done :)

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Documentation
+
+### Guides
```
Comment (Martchus) [inline docs/README.md:3]:
```suggestion
## Guides
```

File: README.md (hunk 4 above)
Comment (Martchus) [reply README.md]: I would prefer to use `nodejs-default` - unless that points to a very old version.

Comment (kraih) [reply README.md]: If i remember correctly the reason i originally set it to nodejs16 was that it was the newest version back then, and we needed some features not available in older releases. Any newer version should be fine now.

**Review by okurz — APPROVED** [review]

**Pull request merged**